### PR TITLE
Add install_data() rename parameter

### LINKED ISSUE #1487: want install_data() variant that allows installing file under a different name
Sometimes one needs to install a file (data), but under a different name than it exists in the source tree, e.g. install `foo.bar.win32` as `foo.bar`.

We can currently hack around that with `configure_file()` and an empty configuration, but would be nice if there was an official way too.

### PR DESCRIPTION
**Pull request by sarum9in**
Closes #1487

This patch adds `rename` parameter to `install_data()`. If specified, length of rename list must be equal to number of sources. Non-basename `rename` arguments are allowed and they are joined with `install_dir`.

Please take a look on `MesonException('Size of rename argument is different from number of sources')`, I think this error message can be formulated better, looking for suggestions.

### CODE REVIEW

**Comment by rilian-la-te** [comment]
O, it is very good. Can you provide this also for targets?

**Comment by sarum9in** [comment]
I would like to implement similar for `executable` and `shared_module` since they are relocatable, but I doubt we want this for normal libraries.

**Comment by rilian-la-te** [comment]
I want this for modules and executables. And for JARs.

**Comment by jpakkane** [comment]
> I want this for modules and executables. And for JARs.

Why is that? Why would the names need to be different in the build directory as opposed to install directory.

**Comment by sarum9in** [comment]
#3246 should be able to cover this I hope.

File: docs/markdown/Reference-manual.md [hunk 1]
```diff
@@ -737,6 +737,31 @@ arguments. The following keyword arguments are supported:
...
+```meson
+install_data(['file1.txt', 'file2.txt'],
+             rename : ['dir1/data.txt', 'dir2/data.txt'],
+             install_dir : 'share/myapp')
+```
+
+Not allowed because of different sources and rename sizes:
+- `install_data('file.txt', rename : ['name1.txt', 'name2.txt'])`
+- `install_data(['file1.txt', 'file2.txt'], rename : 'name.txt')`
+
```
Comment (jpakkane) [inline docs/markdown/Reference-manual.md]: This should be in Installing.md. All examples and usage instructions should be there.

**Comment by sarum9in** [comment]
@jpakkane moved examples to `Installing.md`, also linked them from `install_data()` reference doc for easier navigation.

**Pull request merged**